server_app/user/handler.js

```diff
@@ -0,0 +1,62 @@
+const express = require("express");
+const bodyParser = require("body-parser");
+const userService = require("./service");
+const userValitor = require("./validator");
+
+const router = express.Router();
+router.use(bodyParser.json());
+
+const userLogin = async(req, res) => {
+  try {
+    const username = req.body.username;
+    const password = req.body.password;
+    const sessionId = req.sessionID;
+
+    const validation = userValitor.validateLoginParams({username, password});
+
+    if(!validation.valid) {
+      return res.status(422).send(validation.message);
+    }
+      
+    const userSession = await userService.login({username, password, sessionId});
```

Comment:
Maybe you can build the user session here, and just find the user in the userService using the password encryption.